features/netsocket/nsapi_dns.h

```diff
@@ -243,6 +253,16 @@ void nsapi_dns_reset();
  */
 extern "C" nsapi_error_t nsapi_dns_add_server(nsapi_addr_t addr, const char *interface_name);
 
+/** Get domain name server address by index from the list of servers to query.
+ *
+ *  @param stack    Network stack as target for DNS query.
+ *  @param index  Index of the DNS server, starts from zero.
+ *  @param dns_addr  DNS server host address.
+ *  @param interface_name  Network interface name
+ *  @return         NSAPI_ERROR_OK on success, negative error code on failure.
+ */
+extern "C" nsapi_error_t nsapi_dns_get_server(NetworkStack *stack, uint8_t index, SocketAddress *dns_addr, const char *interface_name);
```

Comment:
@mikaleppanen Should I make the interface_name optional here also ?

Reply:
Also, no need to make optional there, those are not called directly by the users of the DNS API.